Big syntax highlighting update

golf1052 opened the pull request:
- Basically completely rewrote the tmLanguage
  - Converted it to yaml and have a conversion step to json

golf1052 commented:
This fixes #13, #15, #18, and #43

rosshamish commented:
Change looks great. Thanks for hacking on this! Two minor things:
1. I see one failing test in the validation CI. Assuming it just needs another run of `npm run test:update-snapshots` ?
2. Could you add a "Contributing" section that explains that you want to edit the yml tmlanguage, and then run `npm run convert` after? I'd like to prevent future contributors from editing the wrong file and misunderstanding the workflow -- kusto-syntax-highlighting/README.md

And then if you want to add yourself to the Contributors list in the root README, please go ahead :)

golf1052 commented:
Yep completely forgot to rerun the tests after adding null support. Fixed.

I've added a CONTRIBUTING.md with some info on running locally, the inner loop, and useful links to have. I've also moved the testing section from the readme into contributing.

GitHub also automatically lists contributors on so no need to add it in the readme.
